comp/core/workloadmeta/collectors/internal/process/process_collector_test.go

```diff
@@ -0,0 +1,557 @@
+					Uids:         []int32{},
+					Gids:         []int32{},
+					CreationTime: time.UnixMilli(creationTime2).UTC(),
+				},
+			},
+		},
+		{
+			description:     "2 new processes and 2 finishes",
+			configOverrides: map[string]interface{}{},
+			processesToCollectA: map[int32]*procutil.Process{
+				1234: {
+					Pid:     1234,
+					Ppid:    6,
+					NsPid:   2,
+					Name:    "some name",
+					Cwd:     "some_directory/path",
+					Exe:     "test",
+					Comm:    "",
+					Cmdline: []string{"c1", "c2", "c3"},
+					Uids:    []int32{1, 2, 3, 4},
+					Gids:    []int32{1, 2, 3, 4, 5},
+					Stats: &procutil.Stats{
+						CreateTime:  creationTime1,
+						Status:      "",
+						Nice:        0,
+						OpenFdCount: 0,
+						NumThreads:  0,
+						CPUPercent:  nil,
+						CPUTime:     nil,
+						MemInfo:     nil,
+						MemInfoEx:   nil,
+						IOStat:      nil,
+						IORateStat:  nil,
+						CtxSwitches: nil,
+					},
+				},
+				9999: {
+					Pid:     9999,
+					Ppid:    8,
+					NsPid:   3,
+					Name:    "some name 9999",
+					Cwd:     "some_directory/path/for",
+					Exe:     "exe",
+					Comm:    "something",
+					Cmdline: []string{"c1", "c2", "c3", "c4"},
+					Uids:    []int32{},
+					Gids:    []int32{},
+					Stats: &procutil.Stats{
+						CreateTime:  creationTime2,
+						Status:      "",
+						Nice:        0,
+						OpenFdCount: 0,
+						NumThreads:  0,
+						CPUPercent:  nil,
+						CPUTime:     nil,
+						MemInfo:     nil,
+						MemInfoEx:   nil,
+						IOStat:      nil,
+						IORateStat:  nil,
+						CtxSwitches: nil,
+					},
+				},
+			},
+			expectedProcessesA: map[int32]*workloadmeta.Process{
+				1234: {
+					EntityID: workloadmeta.EntityID{
+						Kind: workloadmeta.KindProcess,
+						ID:   "1234",
+					},
+
+					Pid:     1234,
+					Ppid:    6,
+					NsPid:   2,
+					Name:    "some name",
+					Cwd:     "some_directory/path",
+					Exe:     "test",
+					Comm:    "",
+					Cmdline: []string{"c1", "c2", "c3"},
+					Uids:    []int32{1, 2, 3, 4},
+					Gids:    []int32{1, 2, 3, 4, 5},
+					// time is created like this so it has the same precision as the
+					//CreationTime: time.UnixMilli(creationTime1.UnixMilli()).UTC(),
+					CreationTime: time.UnixMilli(creationTime1).UTC(),
+				},
+				9999: {
+					EntityID: workloadmeta.EntityID{
+						Kind: workloadmeta.KindProcess,
+						ID:   "9999",
+					},
+					Pid:          9999,
+					Ppid:         8,
+					NsPid:        3,
+					Name:         "some name 9999",
+					Cwd:          "some_directory/path/for",
+					Exe:          "exe",
+					Comm:         "something",
+					Cmdline:      []string{"c1", "c2", "c3", "c4"},
+					Uids:         []int32{},
+					Gids:         []int32{},
+					CreationTime: time.UnixMilli(creationTime2).UTC(),
+				},
+			},
+			processesToCollectB: map[int32]*procutil.Process{},
+			expectedProcessesB: map[int32]*workloadmeta.Process{
+				1234: nil,
+				9999: nil,
+			},
```

Comment:
If you just need the pids and liveness, lets use a slice of structs with that information or two different slices (live vs dead). I think that makes the test more clear